app/src/main/java/com/android/unio/model/association/AssociationRepositoryFirestore.kt

```diff
@@ -24,7 +24,7 @@ class AssociationRepositoryFirestore @Inject constructor(private val db: Firebas
 
   override fun init(onSuccess: () -> Unit) {
     Firebase.auth.registerAuthStateListener {
-      if (it.currentUser != null) {
+      if (it.currentUser != null && it.currentUser!!.isEmailVerified) {
```

Comment:
Redundant null check

Reply:
My bad, I've told you this but actually these checks are indeed very useful. If they are not in place, null assertion errors get thrown. 
We don't want to assert that the user is null as he can totally be null (for example when he doesn't have an account yet). Nevertheless the onSuccess method should only be called when the user is not null that is checked first, and then that his email is verified (Recall Lazy checks in software construction).